blackjax/inference/trajectory.py

```diff
@@ -199,30 +217,47 @@ def integrate(
             been met.
         step_size
             The step size of the symplectic integrator.
+        initial_energy
+            Initial energy H0 of the HMC step (not to confused with the initial energy of the subtree)
 
         """
 
         def do_keep_integrating(expansion_state):
             """Decide whether we should continue integrating the trajectory"""
-            _, step, _, _, _, is_diverging, has_terminated = expansion_state
+            _, step, _, _, _, _, is_diverging, has_terminated = expansion_state
             return (step < max_num_steps) & ~has_terminated & ~is_diverging
 
         def add_one_state(expansion_state):
             (
                 rng_key,
                 step,
                 proposal,
+                previous_state,
                 trajectory,
                 termination_state,
                 _,
                 _,
             ) = expansion_state
-            _, rng_key = jax.random.split(rng_key)
+            rng_key, proposal_key = jax.random.split(rng_key)
+
+            new_state = integrator(previous_state, direction * step_size)
+            new_proposal, is_diverging = generate_proposal(initial_energy, new_state)
+
+            new_trajectory, sampled_proposal = jax.lax.cond(
```

Comment:
Could we treat the first step separately to avoid going through that `lax.cond` at every step? The code would be cleaner and I assume this is inefficient in JAX.

Reply:
Yeah that's a good idea.

Reply:
With that change now GLOBAL["count"] is 2 instead of 1 in test_compilation.py for NUTS - is this a big issue?

Reply:
As long as it's not a multiple of `num_samples` we're good.